File: C#/Algorithms/Recursion/isPalindrome.cs
using System;

/**
 * Class IsPalindrome
 */
public class IsPalindrome
{
    /**
    * Checks if a string is palindrome using recursion
    *
    * @param String str
    * @return bool
    */
    public bool isPalindromeRecursion(string str)
    {
        if (str.Length < 2) {
            return true;
        }

        if (str[0] != str[str.Length - 1]) {
            return false;
        }

        return isPalindromeRecursion(str.Substring(1, str.Length - 2));
    }
}

File: PHP/Algorithms/String/isPalindrome.php
<?php

/**
 * Checks if a string is palindrome using iteration
 *
 * @param string $string
 * @return bool
 */
function isPalindromeIteration($string)
{
    $length = strlen($string);

    for ($i = 0, $j = $length - 1;$i <= $length / 2 && $j >= $length / 2;$i++, $j--) {
        if ($string[$i] != $string[$j]) {
            return false;
        }
    }

    return true;
}
